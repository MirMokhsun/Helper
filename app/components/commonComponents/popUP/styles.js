import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    popUpWrapper: {
        flex: 1,
        height: '100%',
        justifyContent: "center",
        alignItems: 'center',
    },
    background: {
        position: 'absolute',
        zIndex: -1,
        backgroundColor: 'rgba(0,0,0,0.4)',
        width: '200%',
        height: '200%',
        top: 0,
        left: 0,
    },
    textBtn: {
        fontFamily: "Roboto-Light",
        fontSize: 16,
        textAlign: 'center',
    },
    popUp: {
        padding: 25,
        borderRadius: 10,
        backgroundColor: '#FFFFFF',
        shadowOffset: { width: 0, height: 4 },
        shadowColor: 'rgba(0, 0, 0, 0.15)',
        shadowRadius: 4,
        elevation: 4,
        shadowOpacity: 1,
    },
    tileWrapper: {
        marginBottom: 15
    },
    titleText: {
        lineHeight: 22,
        fontFamily: "Roboto-Light",
        fontSize: 16,
        textAlign: 'center',
        color: '#000756',
    },
    btnWrapper: {
        flexDirection: 'row',
        justifyContent: 'space-evenly'
    },
    btnWhite: {
        width: 118,
        height: 38,
        borderRadius: 20,
        borderWidth: 1,
        borderColor: '#1459D2',
        backgroundColor: '#FFFFFF',
        justifyContent: 'center',
        alignItems: 'center',
    },
    btn: {
        width: 120,
        height: 40,
        borderRadius: 20,
        backgroundColor: '#1459D2',
        justifyContent: 'center',
        alignItems: 'center',
    },

});