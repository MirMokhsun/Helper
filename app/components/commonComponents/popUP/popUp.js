import React from 'react';
import { Text, View, TouchableWithoutFeedback, TouchableOpacity, BackHandler } from 'react-native';
import { styles } from './styles';

export class Popup extends React.Component {
    constructor(props) {
        super(props);
    }

    componentDidMount = () => {
        BackHandler.addEventListener('hardwareBackPressPopup', this.hardwareBackPressPopup);
    }

    UNSAFE_componentWillUnmount = () => {
        BackHandler.removeEventListener('hardwareBackPressPopup', this.hardwareBackPressPopup);
    }

    hardwareBackPressPopup = () => {
        const { backgroundFunction } = this.props;
        backgroundFunction();
    }

    render() {
        const {
            width,              // ширина попапа, обязательна
            backgroundFunction, // функция которая вешаеться на пространство за попапом, закрывающая
            topComponent,       // дополнительный компонент, елси его нет, то можно не чего не вводить
            title,              // основной текст, необязателен
            textBtn,            // текст первой кнопки
            functionBtn,        // функция первой кнопки
            textBtnWhite,       // текст второй кнопки, если ее нет то, false
            functionBtnWhite    // функция второй кнопки, если ее нет то, false
        } = this.props;
        return (
            <TouchableWithoutFeedback testID="closePopupID" accessibilityLabel="closePopupID" onPress={backgroundFunction}>
                <View style={styles.popUpWrapper}>
                    {/* ниже костыль для заднего фона, почему-то связь с брокером показывает задний фон меньше, чем к примеру выход */}
                    <View style={styles.background} /> 
                    <TouchableWithoutFeedback onPress={() => {
                        console.log('click on background popUp')
                    }} >
                        <View style={[styles.popUp, { width }]}>
                            {/* в блоке ниже будет свг или дополнительный текст */}
                            {topComponent || null}
                            {/* ниже тайтл */}
                            {title ? (
                                <View style={styles.tileWrapper}>
                                    <Text style={styles.titleText}>{title}</Text>
                                </View>
                            ) : null}
                            <View style={styles.btnWrapper}>
                                {/* Белая кнопка */}
                                {textBtnWhite ? (
                                    <TouchableOpacity onPress={functionBtnWhite} style={styles.btnWhite} testID="leftBtnPopupID" accessibilityLabel="leftBtnPopupID"    >
                                        <Text style={[styles.textBtn, { color: '#1459D2' }]}>{textBtnWhite}</Text>
                                    </TouchableOpacity>
                                ) : null}
                                {/* Синяя кнопка */}
                                {textBtn ? (
                                    <TouchableOpacity onPress={functionBtn} style={styles.btn} testID="rightBtnPopupID" accessibilityLabel="rightBtnPopupID" >
                                        <Text style={[styles.textBtn, { color: '#FFFFFF' }]}>{textBtn}</Text>
                                    </TouchableOpacity>
                                ) : null}
                                {/*  */}
                            </View>
                        </View>
                    </TouchableWithoutFeedback>
                </View>
            </TouchableWithoutFeedback >
        );
    }
}
