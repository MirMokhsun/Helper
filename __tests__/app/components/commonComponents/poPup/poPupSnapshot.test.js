import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Popup } from '../../../../../app/components/commonComponents/popUP/popUp';

it('create snapshot Popup', () => {
    const renderer = new ShallowRenderer()
    let popup = renderer.render(<Popup />);
    expect(popup).toMatchSnapshot();
});