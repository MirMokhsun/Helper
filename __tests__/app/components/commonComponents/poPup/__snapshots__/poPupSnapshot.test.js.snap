// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot Popup 1`] = `
<TouchableWithoutFeedback
  accessibilityLabel="closePopupID"
  testID="closePopupID"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "height": "200%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "200%",
          "zIndex": -1,
        }
      }
    />
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "elevation": 4,
              "padding": 25,
              "shadowColor": "rgba(0, 0, 0, 0.15)",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
            },
            Object {
              "width": undefined,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-evenly",
            }
          }
        />
      </View>
    </TouchableWithoutFeedback>
  </View>
</TouchableWithoutFeedback>
`;
